sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -435,6 +439,40 @@ def process_pipelineparam(s):
         }
         self.loops_pipeline[group_name]['spec']['params'].append(sep_param)
 
+      if hasattr(sub_group, 'start') and sub_group.start is not None:
+        # start, end, step params should be added as a parameter
+        # isinstance(sub_group.start, dsl.PipelineParam)
+        def process_parameter(parameter):
+          parameter_value = str(parameter)
+          if isinstance(parameter, dsl.PipelineParam):
+            if parameter.op_name:
+              parameter_value = '$(tasks.' + parameter.op_name + '.results.' + sanitize_k8s_name(parameter.name) + ')'
+            else:
+              parameter_value = '$(params.' + parameter.name + ')'
```

Comment:
What if it's a combination of string and pipeline parameter or multiple pipeline parameters?

For instance:
```
start="{{pipelineparam:op=;name=integer}}.{{pipelineparam:op=;name=decimal}}"
```

Reply:
I think start, to, step only can loop over integers. Also, the start parameter can either be pipeline param or string, not both at the same time.

Reply:
if this is necessary, we probably need to convert the pipeline param into string, then use some regular expression to parse the `{{pipelineparam:op=;name=integer}}` syntax back to Tekton params for this use case only. I think to make the compiler less complex, it's better to treat it like python range() where we should use these number as indices rather than decimal inputs.

Reply:
Sounds quite complex indeed. It's not a must for our team, just a quick thought that the general user might want to try that kind of thing.

I think what kfp-tekton could do right now is to try parse the pipelineparam from string. If there is only one and it matches the whole string --- then it's all good and fine. If not --- an error indicating that it's not supported can be raised.